core/src/main/java/be/orbinson/aem/dictionarytranslator/services/impl/CombiningMessageEntryResourceProvider.java

```diff
@@ -80,21 +186,62 @@ public class CombiningMessageEntryResourceProvider extends ResourceProvider<Obje
                     throw new ResourceNotFoundException("Unable to get message entries for language '" + language + "' in dictionary '" + dictionaryPath + "'", e);
                 }
                 messagePerLanguage.put(language, message);
+                if (message != null) {
+                    validateItem(dictionaryResource, language, key, message.getText()).ifPresent(validationMessages::add);
+                }
             }
-           return new ValueMapResource(resourceResolver, path, RESOURCE_TYPE, new ValueMapDecorator(createResourceProperties(path, dictionaryService.isEditableDictionary(dictionaryResource), messagePerLanguage)));
+            return new ValueMapResource(resourceResolver, path, RESOURCE_TYPE, new ValueMapDecorator(createResourceProperties(path, dictionaryService.isEditableDictionary(dictionaryResource), messagePerLanguage, validationMessages)));
         } else {
             throw new ResourceNotFoundException(path, "Unable to get underlying dictionary resource for path '" + dictionaryPath + "'");
         }
     }
 
+    private Optional<ValidationMessage> validateItem(Resource dictionaryResource, String language, String key, String message) {
+        Optional<Resource> conflictingDictionaryResource = dictionaryService.getConflictingDictionary(dictionaryResource, language, key);
+        final ValidationMessage validationMessage;
+        // check severity as well
+        if (conflictingDictionaryResource.isPresent()) {
+            // check if message is different from the one in the dictionary
+            String otherMessage;
+            try {
+                otherMessage = dictionaryService.getMessages(conflictingDictionaryResource.get(), language).get(key).getText();
+            } catch (DictionaryException e) {
+                throw new IllegalStateException("Unable to get message entries for language '" + language + "' in dictionary '" + conflictingDictionaryResource.get().getPath() + "'", e);
+            }
+            if (Objects.equals(otherMessage, message)) {
+                validationMessage = new ValidationMessage(ValidationMessage.Severity.INFO, language, "Conflicting dictionary at {0} for language {1}, it has the same message though", conflictingDictionaryResource.get().getPath(), language);
+            } else {
+                String conflictingDictionaryBasename = dictionaryService.getBasename(conflictingDictionaryResource.get());
+                switch (checkBasenameOverlap(dictionaryService.getBasename(dictionaryResource), conflictingDictionaryBasename)) {
+                    case PARTIAL:
+                        validationMessage = new ValidationMessage(ValidationMessage.Severity.WARNING, language, "Conflicting dictionary at {0} for language {1} with another translation and partially overlapping basenames {2}", conflictingDictionaryResource.get().getPath(), language);
```

Comment:
Same here, it's a warning and not an error, so what will be the effect ?

Reply:
potentially this item is not used (depending on the basename used when referencing)